service/frontend/workflow_handler.go

```diff
@@ -2092,69 +2092,90 @@ func (wh *WorkflowHandler) ListOpenWorkflowExecutions(ctx context.Context, reque
 		return nil, errRequestNotSet
 	}
 
+	maxPageSize := int32(wh.config.VisibilityMaxPageSize(request.GetNamespace()))
+	if request.GetMaximumPageSize() <= 0 || request.GetMaximumPageSize() > maxPageSize {
+		request.MaximumPageSize = maxPageSize
+	}
+
+	namespaceName := namespace.Name(request.GetNamespace())
+	namespaceID, err := wh.namespaceRegistry.GetNamespaceID(namespaceName)
+	if err != nil {
+		return nil, err
+	}
+
+	baseReq := &manager.ListWorkflowExecutionsRequestV2{
+		NamespaceID:   namespaceID,
+		Namespace:     namespaceName,
+		PageSize:      int(request.GetMaximumPageSize()),
+		NextPageToken: request.NextPageToken,
+		Query: fmt.Sprintf(
+			"%s = '%s'",
+			searchattribute.ExecutionStatus,
+			enumspb.WORKFLOW_EXECUTION_STATUS_RUNNING,
+		),
+	}
+
 	if request.StartTimeFilter == nil {
 		request.StartTimeFilter = &filterpb.StartTimeFilter{}
 	}
 
 	if request.StartTimeFilter.EarliestTime == nil || request.StartTimeFilter.EarliestTime.AsTime().IsZero() {
 		request.StartTimeFilter.EarliestTime = timestamppb.New(minTime)
+	} else {
+		baseReq.Query += fmt.Sprintf(
+			"AND %s >= '%s'",
+			searchattribute.StartTime,
+			request.StartTimeFilter.EarliestTime.AsTime().Format(time.RFC3339Nano),
+		)
 	}
 
 	if request.StartTimeFilter.LatestTime == nil || request.StartTimeFilter.LatestTime.AsTime().IsZero() {
 		request.StartTimeFilter.LatestTime = timestamppb.New(maxTime)
+	} else {
+		baseReq.Query += fmt.Sprintf(
+			"AND %s <= '%s'",
+			searchattribute.StartTime,
+			request.StartTimeFilter.LatestTime.AsTime().Format(time.RFC3339Nano),
+		)
 	}
 
 	if request.StartTimeFilter.EarliestTime.AsTime().After(request.StartTimeFilter.LatestTime.AsTime()) {
 		return nil, errEarliestTimeIsGreaterThanLatestTime
 	}
```

Comment:
Let's optimize further the query, and use `BETWEEN ... AND ...`

```suggestion
	earliestTime := request.GetStartTimeFilter().GetEarliestTime()
	latestTime := request.GetStartTimeFilter().GetLatestTime()
	if earliestTime != nil && !earliestTime.AsTime().IsZero &&
		latestTime != nil && !latestTime.AsTime().IsZero {
		fromTime := earliestTime.AsTime().UTC()
		toTime := latestTime.AsTime().UTC()
		if fromTime.After(toTime) {
			return nil, errEarliestTimeIsGreaterThanLatestTime
		}
		baseReq.Query += fmt.Sprintf(
			" AND %s BETWEEN '%s' AND '%s'",
			searchattribute.StartTime,
			fromTime.Format(time.RFC3339Nano),
			toTime.Format(time.RFC3339Nano),
		)
	} else if earliestTime != nil && !earliestTime.AsTime().IsZero {
		baseReq.Query += fmt.Sprintf(
			" AND %s >= '%s'",
			searchattribute.StartTime,
			earliestTime.AsTime().UTC().Format(time.RFC3339Nano),
		)
	} else if latestTime != nil && !latestTime.AsTime().IsZero {
		baseReq.Query += fmt.Sprintf(
			" AND %s <= '%s'",
			searchattribute.StartTime,
			latestTime.AsTime().UTC().Format(time.RFC3339Nano),
		)
	}
```

Reply:
I believe the query BETWEEN .. AND isn't necessary faster than GTE or LTE queries. If that is correct, would there be any reason to separate cases to use the query BETWEEN? Using BETWEEN would require to add extra case in query converter.

Reply:
The query converter already handles the BETWEEN expression. For ES, it's builds one range query instead of two.

Reply:
I see, made changes to use BETWEEN AND!